src/runtime/opencl.cpp

```diff
@@ -704,21 +705,25 @@ WEAK int halide_opencl_device_malloc(void *user_context, buffer_t* buf) {
         return ctx.error;
     }
 
-    size_t size = buf_size(buf);
+    size_t size = buf->size_in_bytes();
     halide_assert(user_context, size != 0);
-    if (buf->dev) {
+    if (buf->device) {
         halide_assert(user_context, validate_device_pointer(user_context, buf, size));
         return 0;
     }
 
-    halide_assert(user_context, buf->stride[0] >= 0 && buf->stride[1] >= 0 &&
-                                buf->stride[2] >= 0 && buf->stride[3] >= 0);
+    for (int i = 0; i < buf->dimensions; i++) {
+        halide_assert(user_context, buf->dim[i].stride >= 0);
+    }
+
 
     debug(user_context)
         << "    Allocating buffer of " << (int)size << " bytes,"
-        << " extents: " << buf->extent[0] << "x" << buf->extent[1] << "x" << buf->extent[2] << "x" << buf->extent[3]
-        << " strides: " << buf->stride[0] << "x" << buf->stride[1] << "x" << buf->stride[2] << "x" << buf->stride[3]
-        << " (" << buf->elem_size << " bytes per element)\n";
+        << " extents: " << buf->dim[0].extent << "x" << buf->dim[1].extent
```

Comment:
What do you think about making the printer (runtime/printer.h) just accept halide_buffer_t and do something reasonable?